alacritty_terminal/src/tty/unix.rs

```diff
@@ -212,12 +212,9 @@ pub fn from_fd(config: &Options, window_id: u64, master: OwnedFd, slave: OwnedFd
     };
 
     // Setup child stdin/stdout/stderr as slave fd of PTY.
-    // Ownership of fd is transferred to the Stdio structs and will be closed by them at the end of
-    // this scope. (It is not an issue that the fd is closed three times since File::drop ignores
-    // error on libc::close.).
-    builder.stdin(unsafe { Stdio::from_raw_fd(slave_fd) });
-    builder.stderr(unsafe { Stdio::from_raw_fd(slave_fd) });
-    builder.stdout(unsafe { Stdio::from_raw_fd(slave_fd) });
+    builder.stdin(rustix_openpty::rustix::io::dup(&slave)?);
+    builder.stderr(rustix_openpty::rustix::io::dup(&slave)?);
+    builder.stdout(slave);
```

Comment:
The slave FDs were previously closed in the child, is there a reason why you're not doing this anymore?

Reply:
it's all the same as before, it's just `fd`'s are valid in each call so the `assert` in `drop`  doesn't get triggered instead of closing multiple times.